File: hp-server/src/main/java/net/hserver/hplite/domian/bean/ResDeviceInfo.java
package net.hserver.hplite.domian.bean;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ResDeviceInfo {

    private String deviceId;

    private String desc;

    private boolean online;
    public ResDeviceInfo(String deviceId, String desc, boolean online) {
        this.deviceId = deviceId;
        this.desc = desc;
        this.online = online;
    }
    private MemoryInfo memoryInfo;
}

File: hp-server-golang/bean/res_device_info.go
package bean

type ResDeviceInfo struct {
	UserId int `json:"userId"`

	Username string `json:"username"`

	UserDesc string `json:"userDesc"`

	DeviceId string `json:"deviceId"`

	Desc string `json:"desc"`

	Online bool `json:"online"`

	MemoryInfo *MemoryInfo `json:"memoryInfo"`
}

func NewResDeviceInfo(deviceId string, desc string, online bool) *ResDeviceInfo {
	return &ResDeviceInfo{
		DeviceId: deviceId,
		Desc:     desc,
		Online:   online,
	}
}
